Add v0.10-alpha release notes (#980)
spiceai/spiceai: docs/release_notes/v0.10-alpha.md

**JDBC-Client Connectivity**: Improved connectivity for JDBC clients, like Tableau.

**User Experience Improvements**:

- Friendlier error messages across the board to make debugging and development better.
- Added a `spice login postgres` command, streamlining the process for connecting to PostgreSQL databases.
- Added PostgreSQL connection verification and connection string support, enhancing usability for PostgreSQL users.

**Grafana Dashboard**: Improving the ability to monitor Spice deployments, a standard Grafana dashboard is now available.

## Contributors

- @phillipleblanc
- @mitchdevenport
- @Jeadie
- @ewgenius
- @sgrebnov
- @y-f-u
- @lukekim
- @digadeesh

## New in this release

- **Fixes** Gracefully handle Arrow Flight DoExchange connection resets
- **Adds** Grafana Dashboard
- **Adds** Flight SQL CommandGetTableTypes Command support (improves JDBC-client connectivity)
- **Adds** Friendlier error messages
- **Adds** `spice login postgres` command
- **Adds** PostgreSQL connection verification
- **Adds** PostgreSQL connection string support

> - **Adds** PostgreSQL TLS connection support

Author: ```suggestion
```

# Spice v0.10-alpha (Mar 27, 2024)

Announcing the release of Spice.ai v0.10-alpha! 🎉

The Spice.ai v0.10-alpha release focused on additions and updates to improve stability, usability, and the overall Spice developer experience.

## Highlights in v0.10-alpha

**Public Bucket Support for S3 Data Connector**: The S3 Data Connector now supports public buckets in addition to buckets requiring an access id and key.

**JDBC-Client Connectivity**: Improved connectivity for JDBC clients, like Tableau.

**User Experience Improvements**:

- Friendlier error messages across the board to make debugging and development better.
- Added a `spice login postgres` command, streamlining the process for connecting to PostgreSQL databases.

> - Introduced PostgreSQL connection verification, connection string support, TLS connection support, enhancing security, and usability for PostgreSQL users.

Author: ```suggestion
- Added PostgreSQL connection verification and connection string support, enhancing usability for PostgreSQL users.
```

**Grafana Dashboard**: Improving the ability to monitor Spice deployments, a standard Grafana dashboard is now available.

## Contributors

- @phillipleblanc
- @mitchdevenport
- @Jeadie
- @ewgenius
- @sgrebnov
- @y-f-u
- @lukekim
- @digadeesh

## New in this release

- **Fixes** Gracefully handle Arrow Flight DoExchange connection resets

> - **Adds** Public bucket support to S3 Data Connector

Author: ```suggestion
```

- **Adds** Grafana Dashboard
- **Adds** Flight SQL CommandGetTableTypes Command support (improves JDBC-client connectivity)
- **Adds** Friendlier error messages
- **Adds** `spice login postgres` command
- **Adds** PostgreSQL connection verification
- **Adds** PostgreSQL connection string support

> - **Adds** PostgreSQL TLS connection support

Reviewer: ```suggestion
```
